lib/test.d.ts

```diff
@@ -0,0 +1,233 @@
+import type { EventEmitter } from 'events';
+import type mockProperty from 'mock-property';
+
+import type {
+    AssertOptions,
+    TestCase,
+    TestOptions,
+} from '../';
+import type {
+    Operator,
+} from './results';
+
+declare class Test extends EventEmitter {
+    constructor(name: string, opts: TestOptions, cb: TestCase);
+    constructor(name: string, opts: TestOptions);
+    constructor(name: string, cb: Test.Callback);
+    constructor(name: string);
```

Comment:
Unless you're documenting each overload separately, it's equivalent to inline them all by declaring later parameters optional:

```suggestion
    constructor(name: string, opts?: TestOptions, cb?: TestCase);
```

Fun fact: [`plugin:@typescript-eslint/unified-signatures`](https://typescript-eslint.io/rules/unified-signatures) would catch this. Since you're already using ESLint I'd definitely recommend adding in typescript-eslint for at least this `.d.ts` file.

Reply:
oh hm. are those really the same? because usually "optional" doesn't mean you can omit any of them, but only that you can omit the last N options.

Reply:
Oh, you're right, my suggestion snippet was wrong. I missed a line. It should be...

```suggestion
    constructor(name: string, cb: TestCase);
    constructor(name: string, opts?: TestOptions, cb?: TestCase);
```

Sorry 🙂